### PR TITLE
Fix old editor settings/issues

### PR DESCRIPTION
**Pull request by mkrueger**
Fixes some issues I saw using the old editor.

Fixes VSTS #915554 - Old editor adds BOM to file on saving

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs [hunk 1]
```diff
@@ -76,6 +76,7 @@ protected override async Task OnInitialize (ModelDescriptor modelDescriptor, Pro
 		{
 			fileDescriptor = modelDescriptor as FileDescriptor;
 			await base.OnInitialize (modelDescriptor, status);
+			Encoding = fileDescriptor.Encoding;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs]: Why is this needed? Was this supposed to be `textEditor.Encoding`? If so, we need to override `OnEncodingChanged` and set it from there.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs [hunk 2]
```diff
@@ -87,6 +88,7 @@ protected override async Task<Control> OnGetViewControlAsync (CancellationToken
 
 				await Init (editor, impl);
 				HasUnsavedChanges = impl.IsDirty;
+				await UpdateStyleParent (Owner, editor.MimeType, default);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs]:
```suggestion
				await UpdateStyleParent (Owner, editor.MimeType, editorOptionsUpdateCancellationSource.Token);
```

And also initialize the CancellationTokenSource regardless of update being called.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs]: It's needed because it needs to be null if not set.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs (hunk 2 above)
Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs]: Needs to run on GetView.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs]: Well, if we get an update while the view is constructed, we will have conflicting data being set.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs]: hm, in any case it's the wrong token OnGetViewControlAsync seems to get a token - wrong sorting of parameters btw.
